queries.sql

```diff
@@ -1,3 +1,183 @@
+SELECT a.name
+FROM animals a
+JOIN owners o ON a.owner_id = o.id
+WHERE o.full_name = 'Dean Winchester' AND a.escape_attempts = 0;
+
+-- Who owns the most animals?
+SELECT o.full_name, COUNT(*) AS animal_count
+FROM owners o
+JOIN animals a ON o.id = a.owner_id
+GROUP BY o.full_name
+ORDER BY animal_count DESC
+LIMIT 1;
+
+-- Write queries to answer the following:
+-- Who was the last animal seen by William Tatcher?
+SELECT animals.name
+FROM visits
+JOIN vets ON visits.vet_id = vets.id
+JOIN animals ON visits.animal_id = animals.id
+WHERE vets.name = 'Vet William Tatcher'
+ORDER BY visits.visit_date DESC
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+  SELECT DISTINCT animals.name AS animal_name
+  FROM visits
+  LEFT JOIN animals ON animals.id = visits.animal_id
+  LEFT JOIN vets ON vets.id = visits.vet_id
+  WHERE vets.name = 'Stephanie Mendez';
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT vet.name AS vet_name, species.name AS species_name FROM vets vet 
+LEFT JOIN  specializations ON specializations.vet_id = vet.id
+LEFT JOIN species ON specializations.species_id = species_id
+ORDER BY vet.name;
+
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name AS animal_name
+FROM animals a
+JOIN visits v ON v.animal_id = a.id
+JOIN vets vet ON vet.id = v.vet_id
+WHERE vet.name = 'Stephanie Mendez'
+  AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+  -- What animal has the most visits to vets?
+SELECT a.name AS animal_name, COUNT(*) AS visit_count
+FROM animals a
+JOIN visits v ON v.animal_id = a.id
+GROUP BY a.id
+ORDER BY visit_count DESC
+LIMIT 1;
+
+    --  Who was Maisy Smith's first visit?
+    SELECT a.name AS animal_name, v.visit_date AS visit_date FROM visits v
+    JOIN animals a ON a.id = v.animal_id
+    JOIN vets vet ON vet_id = v.vet_id
+    WHERE vet.name = ' Maisy Smith'
+    ORDER BY v.visit_date ASC
+    LIMIT 1;
+
+ -- Details for the most recent visit: animal information, vet information, and date of visit.
+ SELECT a.name AS animal_name, vet.name AS vet_name, v.visit_date
+FROM visits v
+JOIN animals a ON a.id = v.animal_id
+JOIN vets vet ON vet.id = v.vet_id
+ORDER BY v.visit_date DESC
+LIMIT 1;
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*) AS count
+FROM visits v
+JOIN animals a ON a.id = v.animal_id
+JOIN vets vet ON vet.id = v.vet_id
+LEFT JOIN specializations spec ON spec.vet_id = vet.id AND spec.species_id = a.species_id
+WHERE spec.id IS NULL;
+
+-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+SELECT species.name AS specialty, COUNT(*) AS number_of_visits
+FROM visits
+JOIN vets ON visits.vet_id = vets.id
+JOIN animals ON visits.animal_id = animals.id
+JOIN species ON animals.species_id = species.id
+WHERE vets.name = 'Maisy Smith'
+GROUP BY species.name
+ORDER BY number_of_visits DESC
```

Comment:
- Good job done so far, but you did not get an answer to this query. Please fix this and attach a screenshot to the correct answer in your PR 🙏.
 ```SQL
SELECT species.name, count(*)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN species ON animals.species_id = species.id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE vet.name = 'Maisy Smith'
  GROUP BY species.name
```